arkit/server.py

```diff
@@ -0,0 +1,256 @@
+        self._tap_action_finished = False
+        self._animation_triggerd = False
+        self._trigger_message = False
+        self._message = ""
+
+        if USE_LOGGING:
+            cozmo.setup_basic_logging()
+        if USE_VIEWER:
+            cozmo.connect_with_tkviewer(self.run)
+
+        else:
+            try:
+                cozmo.connect(self.run)
+            except cozmo.ConnectionError as e:
+                sys.exit("A connection error occurred: %s" % e)
+
+    async def run(self, coz_conn):
+        """The run method runs once Cozmo is connected."""
+        await self.set_up_cozmo(coz_conn)
+
+        while True:
+            await asyncio.sleep(0)
+        pass
+
+    async def set_up_cozmo(self, coz_conn):
+        asyncio.set_event_loop(coz_conn._loop)
+        self._robot = await coz_conn.wait_for_robot()
+
+        global serverT
+        serverT = threading.Timer(0, self.run_server).start()
+        await self._robot.set_lift_height(0, duration=0.5).wait_for_completed()
+        await self._robot.play_anim("anim_launch_wakeup_02").wait_for_completed()
+        await self._robot.play_anim("anim_hiking_lookaround_01").wait_for_completed()
+        await self._robot.play_anim("anim_hiking_lookaround_03").wait_for_completed()
+        await self._robot.play_anim("anim_hiking_lookaround_02").wait_for_completed()
+        await self._robot.go_to_pose(Pose(0, 0, 0, angle_z=degrees(0)), relative_to_robot=True).wait_for_completed()
+        await self._robot.set_head_angle(cozmo.util.Angle(degrees=0)).wait_for_completed()
+        self._robot.world.add_event_handler(cozmo.objects.EvtObjectTapped, self.on_object_tapped)
+
+        """Uncomment to start in a sepcific sequence
+        self._round = 2
+        """
+        while self._round < 3:
+            print("self._round: %s" %self._round)
+            await self.find_cube()
+            print("selftap finish: %s" %self._tap_action_finished)
+            if self._tap_action_finished:
+                self._round = self._round + 1
+                self._tap_action_finished = False
+            if DEBUG_MODE:
+                if self._round == 3:
+                    self._round = 0
+
+    async def find_cube(self):
+        cubes = None
+        if self._cube:
+            self._cube.stop_light_chaser()
+            self._cube.set_lights_off()
+        look_around = self._robot.start_behavior(cozmo.behavior.BehaviorTypes.LookAroundInPlace)
+        try:
+            cubes = await self._robot.world.wait_until_observe_num_objects(1, cozmo.objects.LightCube)
+            self._cube = cubes[0]
+            look_around.stop()
+        except TimeoutError:
+            print("Didn't find a cube :-(")
+            return False
+        if self._round < 2:
+            tap_anim = random.choice(["anim_pyramid_reacttocube_happy_high_01", "anim_speedtap_winround_intensity02_02"])
+            await self._robot.play_anim(tap_anim).wait_for_completed()
```

Comment:
As per above, use of play_anim with specific animation name is fragile and could break with a future app version.